internal/state/txindex/indexer_service.go

```diff
@@ -78,19 +78,8 @@ func (is *IndexerService) OnStart() error {
 					txResult := msg2.Data().(types.EventDataTx).TxResult
 
 					if err = batch.Add(&txResult); err != nil {
-						is.Logger.Error(
-							"failed to add tx to batch",
-							"height", height,
-							"index", txResult.Index,
-							"err", err,
-						)
-
-						if is.terminateOnError {
-							if err := is.Stop(); err != nil {
-								is.Logger.Error("failed to stop", "err", err)
-							}
-							return
-						}
+						is.handleError(err, height, txResult.Index)
+						return
```

Comment:
this is incorrect behavior

before it used to be `if is.terminateOnError { return }`

Reply:
I reverted these changes.